test/test_optim.py

```diff
@@ -820,6 +866,10 @@ def test_multi_tensor_optimizers(self):
             (optim.Adagrad, dict(weight_decay=1)),
         ]
         self._test_derived_optimizers(optimizer_pairs_with_flags, "foreach")
+        # TODO(crcrpar): Investigate ROCm failure in
+        # https://github.com/pytorch/pytorch/actions/runs/4910476500/jobs/8767913127
+        if not TEST_WITH_ROCM:
```

Comment:
This should be investigated before landing the PR

Reply:
Even we do not have access to ROCm machines, as the CI is provided from AMD 😛. 

The first thing I would try here is seeing if adding an empty param to the param list in _test_derived_optimizers (the merge I recommend) would still fail ROCm. If it still does, we can involve the AMD people on Slack to get help debugging.

Reply:
Also, this issue is likely relevant just FYI https://github.com/pytorch/pytorch/issues/69698.

I'd still recommend merging with the other test.